java/statestore/src/main/java/sleeper/statestore/s3/S3PartitionStore.java

```diff
@@ -266,9 +252,14 @@ public void initialise(List<Partition> partitions) throws StateStoreException {
     }
 
     private void setPartitions(List<Partition> partitions) throws StateStoreException {
+        // Validate that there is no current revision id
+        RevisionId oldRevisionId = s3RevisionUtils.getCurrentPartitionsRevisionId();
+        if (null != oldRevisionId && !oldRevisionId.getRevision().equals(FIRST_REVISION)) {
+            throw new StateStoreException("Dynamo should not contain current revision id; found " + oldRevisionId);
+        }
```

Comment:
This check is not needed. It seems to be a merge conflict with the changes from PR #1362, where this check was removed in the original code and replaced with logic in DelegatingStateStore.